app/assets/javascripts/discourse/app/components/sidebar/tags-section.js

```diff
@@ -7,7 +7,16 @@ export default class SidebarTagsSection extends GlimmerComponent {
   @cached
   get sectionLinks() {
     return this.currentUser.trackedTags.map((trackedTag) => {
-      return new TagSectionLink({ tag: trackedTag });
+      return new TagSectionLink({
+        tag: trackedTag,
+        topicTrackingState: this.topicTrackingState,
+      });
+    });
+  }
+
+  willDestroy() {
+    this.sectionLinks.forEach((sectionLink) => {
+      sectionLink.teardown();
```

Comment:
Hmm if `this.currentUser.trackedTags` changes and `sectionLinks` is re-computed, is there anything that calls `teardown()` on the array elements of `sectionLinks` before it's re-computed?

(we may have the same problem with `categories-section`, it didn't occur to me when I reviewed the PR yesterday)

Reply:
@OsamaSayegh Good catch! I've fixed this PR and will follow up with another PR for the other sections.